### PR TITLE
Additional Changes on table filters

### PR DESCRIPTION
**Pull request by priyesh777**
## Addresses:
- https://github.com/idmc-labs/Helix2.0/issues/61

## Depends on:
- https://github.com/idmc-labs/helix-server/pull/221

## Changes:
- Add additional changes to the Contextual updates filter

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [x] permission checks
- [x] translations

### CODE REVIEW

File: src/views/Events/EventsFilter/index.tsx [hunk 1]
```diff
@@ -169,7 +169,7 @@ function EventsFilter(props: EventsFilterProps) {
     } = useQuery<EventOptionsForFiltersQuery>(EVENT_OPTIONS);
 
     const violenceOptions = data?.violenceList?.results;
-    const disasterCategoryOptions = data?.disasterCategoryList?.results;
+    const disasterCategoryOptions = data?.disasterSubTypeList?.results;
```
Comment (tnagorra) [inline src/views/Events/EventsFilter/index.tsx:172]:
```suggestion
    const disasterSubTypeOptions = data?.disasterSubTypeList?.results;
```

We will need to modify the form and query itself!

File: src/views/Extraction/ExtractionFilters/index.tsx [hunk 2]
```diff
@@ -608,7 +608,7 @@ function ExtractionFilters(props: ExtractionFiltersProps) {
                 />
                 <MultiSelectInput
                     options={data?.figureCategoryList?.results}
-                    label="Figure Category Type"
+                    label="Figure Category"
```
Comment (tnagorra) [inline src/views/Extraction/ExtractionFilters/index.tsx:611]:
```suggestion
                    label="Figure Categories"
```

File: src/views/Extraction/ExtractionFilters/index.tsx [hunk 3]
```diff
@@ -597,7 +597,7 @@ function ExtractionFilters(props: ExtractionFiltersProps) {
             <Row>
                 <MultiSelectInput
                     options={data?.figureCategoryType?.enumValues}
-                    label="Figure Category"
+                    label="Figure Category Type"
```
Comment (tnagorra) [inline src/views/Extraction/ExtractionFilters/index.tsx:600]:
```suggestion
                    label="Figure Category Types"
```

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: src/views/Extraction/ExtractionFilters/index.tsx [hunk 4]
```diff
@@ -609,7 +609,7 @@ function ExtractionFilters(props: ExtractionFiltersProps) {
                 />
                 <MultiSelectInput
                     options={data?.figureCategoryList?.results}
-                    label="Figure Category Type"
+                    label="Figure Categorys"
```
Comment (AdityaKhatri) [inline src/views/Extraction/ExtractionFilters/index.tsx:612]:
```suggestion
                    label="Figure Categories"
```

File: src/views/Events/EventsFilter/index.tsx [hunk 5]
```diff
@@ -169,7 +202,24 @@ function EventsFilter(props: EventsFilterProps) {
...
+    const disasterSubTypeOptions = data?.disasterCategoryList?.results?.flatMap((disasterCategory) => (
+        disasterCategory.subCategories?.results?.flatMap((disasterSubCategory) => (
+            disasterSubCategory.types?.results?.flatMap((disasterType) => (
+                disasterType.subTypes?.results?.map((disasterSubType) => ({
+                    ...disasterSubType,
+                    disasterTypeId: disasterType.id,
+                    disasterTypeName: disasterType.name,
+                    disasterSubCategoryId: disasterSubCategory.id,
+                    disasterSubCategoryName: disasterSubCategory.name,
+                    disasterCategoryId: disasterCategory.id,
```
Comment (AdityaKhatri) [inline src/views/Events/EventsFilter/index.tsx:216]: Fix indentation

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

File: src/views/Extraction/ExtractionFilters/index.tsx (hunk 4 above)
Comment (tnagorra) [reply src/views/Extraction/ExtractionFilters/index.tsx:612]: oh no!

File: src/views/Events/EventsFilter/index.tsx (hunk 5 above)
Comment (tnagorra) [reply src/views/Events/EventsFilter/index.tsx:216]: What is the problem sir?

Comment (AdityaKhatri) [reply src/views/Events/EventsFilter/index.tsx:216]: bad indentation sir

**Pull request merged**